src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -105,14 +105,16 @@ void HSHomeObject::register_homestore_metablk_callback() {
     HomeStore::instance()->meta_service().register_handler(
         _shard_meta_name,
         [this](homestore::meta_blk* mblk, sisl::byte_view buf, size_t size) { on_shard_meta_blk_found(mblk, buf); },
-        [this](bool success) { on_shard_meta_blk_recover_completed(success); }, true);
+        [this](bool success) { on_shard_meta_blk_recover_completed(success); }, true,
+        std::optional< meta_subtype_vec_t >(meta_subtype_vec_t{_pg_meta_name}));
 
     HomeStore::instance()->meta_service().register_handler(
         _pg_meta_name,
         [this](homestore::meta_blk* mblk, sisl::byte_view buf, size_t size) {
             on_pg_meta_blk_found(std::move(buf), voidptr_cast(mblk));
         },
-        nullptr, true);
+        // TODO: move "repl_dev" to homestore::repl_dev and "index" to homestore::index
+        nullptr, true, std::optional< meta_subtype_vec_t >(meta_subtype_vec_t{"repl_dev", "index"}));
```

Comment:
apply dependency to index is actually unnecessary and will slow down the recovery time.